ouroboros-praos/src/consensus/mod.rs

```diff
@@ -0,0 +1,349 @@
+use malachite::platform_64::Limb;
+use malachite::Natural;
+use ouroboros::ledger::{issuer_vkey_to_pool_id, PoolId, PoolInfo};
+use ouroboros::validator::Validator;
+use pallas_crypto::hash::{Hash, Hasher};
+use pallas_crypto::vrf::{
+    VrfProof, VrfPublicKey, VRF_PROOF_HASH_SIZE, VRF_PROOF_SIZE, VRF_PUBLIC_KEY_SIZE,
+};
+use pallas_math::math::{ExpOrdering, FixedDecimal, FixedPrecision};
+use pallas_primitives::babbage;
+use std::ops::Deref;
+use std::sync::LazyLock;
+use tracing::{error, span, trace, warn};
+
+/// The certified natural max value represents 2^256 in praos consensus
+static CERTIFIED_NATURAL_MAX: LazyLock<FixedDecimal> = LazyLock::new(|| {
+    FixedDecimal::from_str(
+        "1157920892373161954235709850086879078532699846656405640394575840079131296399360000000000000000000000000000000000",
+        34,
+    )
+        .expect("Infallible")
+});
+
+/// Validator for a block using praos consensus.
+pub struct BlockValidator<'b> {
+    header: &'b babbage::Header,
+    pool_info: &'b dyn PoolInfo,
+    epoch_nonce: &'b Hash<32>,
+    // c is the ln(1-active_slots_coeff). Usually ln(1-0.05)
+    c: &'b FixedDecimal,
+}
+
+impl<'b> BlockValidator<'b> {
+    // TODO: This function is currently only used in tests. Remove warning suppression in the future
+    #[allow(unused)]
```

Comment:
What about making it public instead? That should clear the warning.

Reply:
right. This is a bug as it should be public anyway.

Reply:
fixed